addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -742,51 +742,77 @@ var SnippetsMenu = Widget.extend({
             return $elem.parent().width() === $elem.outerWidth(true);
         }
 
+        // Firstly, add a dropzone after the clone
+        var $clone = $('.oe_drop_clone');
+        var $drop = zone_template.clone();
+        var neighbor = $clone.prev()[0] || $clone.next()[0];
+
+        if (neighbor) {
+            var $neighbor = $(neighbor);
+            var css = window.getComputedStyle(neighbor);
+            var parentCss = window.getComputedStyle($neighbor.parent()[0]);
+            var float = css.float || css.cssFloat;
+            var parentDisplay = parentCss.display;
+            var parentFlex = parentCss.flexDirection;
+            if (float === 'left' || float === 'right' || (parentDisplay === 'flex' && parentFlex === 'row')) {
+                $drop.css('float', float);
+                if (!isFullWidth($neighbor)) {
+                    $drop.addClass('oe_vertical').css('height', Math.max($neighbor.outerHeight(), 30));
+                }
+            }
+        }
+        $clone.after($drop);
+
         if ($selectorChildren) {
             $selectorChildren.each(function () {
                 var $zone = $(this);
                 var css = window.getComputedStyle(this);
-                var parentCss = window.getComputedStyle($zone.parent()[0]);
                 var float = css.float || css.cssFloat;
-                var parentDisplay = parentCss.display;
-                var parentFlex = parentCss.flexDirection;
+                var display = css.display;
+                var flex = css.flexDirection;
                 var $drop = zone_template.clone();
-
-                $zone.append($drop);
-                var node = $drop[0].previousSibling;
-                var test = !!(node && ((!node.tagName && node.textContent.match(/\S/)) ||  node.tagName === 'BR'));
-                if (test) {
-                    $drop.addClass('oe_vertical').css({
-                        height: parseInt(window.getComputedStyle($zone[0]).lineHeight),
-                        float: 'none',
-                        display: 'inline-block',
-                    });
-                } else if (float === 'left' || float === 'right' || (parentDisplay === 'flex' && parentFlex === 'row')) {
-                    $drop.css('float', float);
-                    if (!isFullWidth($zone)) {
-                        $drop.addClass('oe_vertical').css('height', Math.max(Math.min($zone.outerHeight(), $zone.children().last().outerHeight()), 30));
```

Comment:
See that I replaced `$zone.children().last().outerHeight()` by '$lastChild.outerHeight()' here because the last child was always the drop zone and we dont want it in this case.